### DISCUSSION TITLE
[HELP] ChildWindow size

### DISCUSSION THREAD

**Discussion by yzzuf** [ocornut/imgui#5389] [New Users Build/Link/Run issues ONLY!]
I'm making a visual in imgui and I want to make the content align with the square, **but it's going over the limit** (I believe it's the scroll)

![example](https://user-images.githubusercontent.com/42506744/173215014-f7032788-d20c-472d-9de1-6fd46b91ce75.png)
)

**Comment by rokups** [comment]
It is not clear what you are doing precisely. Also there is no question. Please explain better.

**Reply by yzzuf** [reply]
The content is passing from BeginChild, it was supposed to stay within the limit. This problem started to appear only when using the scroll

**What I need to know is how to make the scroll not go over BeginChild, make it adjust to the defined limit?**

I've attached some screenshots to understand better, it turns out it was for the content to adjust to ImVec2(300, 205)

![Screenshot_8](https://user-images.githubusercontent.com/42506744/173350288-b808340c-7a62-4a70-a3b6-15a84fbb01f6.png)
![Screenshot_7](https://user-images.githubusercontent.com/42506744/173350291-09e15509-b9da-406e-a486-3d1b76656dc0.png)
![Screenshot_10](https://user-images.githubusercontent.com/42506744/173350858-f57c989e-aa7e-44c9-8805-434638aa376c.png)
![Screenshot_9](https://user-images.githubusercontent.com/42506744/173350860-20e41b12-6e1a-418e-98b5-a0935f0a9474.png)

**Reply by rokups** [reply]
You do plenty of manual rendering and it is hard to track down what is going on from screenshots. Though i have a suspicion it is a mistake in your code. You can open demo > tools > metrics and inspect dimensions of the child window. I bet you will be surprised to find out child window is bigger than you thought.

**Reply by yzzuf** [reply]
i used
`ImGui::GetForegroundDrawList()->AddRect(ImGui::GetItemRectMin(), ImGui::GetItemRectMax(), IM_COL32(255, 255, 255, 255));`
to draw the child, but idk how to adjust

![image](https://user-images.githubusercontent.com/42506744/173355602-9eb3f15c-11c6-47dd-9482-0e2ef68fd1b3.png)

**Reply by yzzuf** [reply]
I tried some changes but unfortunately the top doesn't go down, it just grows down and to the sides
